### PR TITLE
add girder_plugins argument to ansible role

### PR DESCRIPTION
**Pull request by opadron**
Here's my initial attempt to address #2271.

I'm not too familiar with the role, or how to go about testing my changes, so I probably need a little guidance to put the finishing touches on this.

I hope folks find this helpful.

### CODE REVIEW

**Comment by danlamanna** [comment]
> I'm not too familiar with the role, or how to go about testing my
> changes, so I probably need a little guidance to put the finishing
> touches on this.
The easiest way is to probably fiddle with
devops/ansible/examples/girder-dev-environment/site.yml and use vagrant
up from Girder root.

> args:
>   chdir: "{{ girder_path }}"
I think this can be removed.

> when: girder_web
This can be removed too, server-side plugins should be able to be installed
as well.

> girder_plugins: []
I'm not sure if this should be a list of strings or a list of
dictionaries, so the user can specify "path" and "flags". For example a
user might want to girder-install plugin --symlink path (or any of the
other flags it takes).

**Comment by opadron** [comment]
@danlamanna What do you think about making the `girder_plugins` option a list of `string|dict` objects.

i.e.:

```YAML
girder_plugins:
  - path: "/path/to/girder/plugin"
    symlink: no # the default

  - "/path/to/girder/plugin"  # same as above

  - path: "/other/path"
    symlink: yes
    other_options????
```

**Comment by danlamanna** [comment]
> @danlamanna What do you think about making the `girder_plugins` option
> a list of `string|dict` objects.

I think it would be nice assuming the YAML logic for handling
strings/dicts interchangeably isn't too complicated.

I'm torn on exposing attributes like symlink as opposed to an args/flags
string where the caller passes something like:
- { "path": "/some/path", "args": "--symlink" }

The rationale is that with this level of interop we don't have the
maintenance burden of altering the ansible role every time a flag gets
added. The flipside is that the girder-install API changes pretty
rarely, so maybe it's not that big a deal.

File: devops/ansible/roles/girder/tasks/main.yml [hunk 1]
```diff
@@ -59,6 +59,10 @@
 - include: daemon.yml
   when: girder_daemonize
 
+- name: Install Girder Plugins
+  command: "{{ girder_install_executable|default('girder-install') }} plugin {{ item.args|default('') }} {{ item.path }}"
```
Comment (danlamanna) [inline devops/ansible/roles/girder/tasks/main.yml]: After I changed `item.path` to `item.path|default(item)`, this example usage worked for me:
```
    girder_plugins:
      - path: "{{ ansible_user_dir }}/other_plugin"
        args: "--symlink"
      - "{{ ansible_user_dir }}/myplugin"
```

Comment (opadron) [reply devops/ansible/roles/girder/tasks/main.yml]: ooh, slick!  With that change, you can also preface the string with your args.

```YAML
girder_plugins:
  - "--symlink {{ ansible_user_dir }}/myplugin"
```
I'll leave it to others to decide if that's desirable or not! 😝

**Comment by opadron** [comment]
@danlamanna LGTY?

**Pull request merged**